src/streamlink/stream/hls_playlist.py

```diff
@@ -238,9 +238,15 @@ def parse(self, data):
         self.state = {}
         self.m3u8 = M3U8()
 
-        for lineno, line in enumerate(filter(bool, data.splitlines())):
-            self.parse_line(lineno, line)
+        lines = enumerate(filter(bool, data.splitlines()))
```

Comment:
You don't need to use enumerate here if you are just ignoring the enumeration :)

Reply:
Perhaps you are using `enumerate` where it would be more suitable to use `iter`. Of course, in Python 3, `filter` returns a `filter` object which it's iterable, and you could use `itertools.ifilter` in Python 2 to make it a generator. 

I assume you tried using `re.finditer` based on `_line_re_`, but didn't find any improvement?

Reply:
Does this still have the same improvement for you as the last version?

Reply:
Right, the best solution is with 'cStringIO' and I'll use it in my version.

Reply:
You'll experience problems if you encounter a playlist with an non-ascii characters using that. 

You could try
```python
if is_py2:
    try:
        from cStringIO import StringIO as BytesIO
    except ImportError:
        from StringIO import StringIO as BytesIO
else:
    from io import BytesIO

...

lines = (l.decode("utf8") for l in BytesIO(data.encode("utf8")
```